mm2src/coins/utxo/utxo_builder/utxo_coin_builder.rs

```diff
@@ -657,16 +660,18 @@ pub trait UtxoCoinBuilderCommonOps {
         Option<UtxoSyncStatusLoopHandle>,
         Option<AsyncMutex<AsyncReceiver<UtxoSyncStatus>>>,
     ) {
-        if self.conf()["enable_spv_proof"].as_bool().unwrap_or(false) && !self.activation_params().mode.is_native() {
+        if self.spv_conf().is_some() && !self.activation_params().mode.is_native() {
             let (sync_status_notifier, sync_watcher) = channel(1);
-            (
+            return (
                 Some(UtxoSyncStatusLoopHandle::new(sync_status_notifier)),
                 Some(AsyncMutex::new(sync_watcher)),
-            )
-        } else {
-            (None, None)
+            );
         }
+
+        (None, None)
     }
+
+    fn spv_conf(&self) -> Option<SPVConf> { serde_json::from_value::<SPVConf>(self.conf()["spv_conf"].clone()).ok() }
```

Comment:
I believe I asked you to use this instead
```rust
fn spv_conf(&self) -> UtxoConfResult<Option<SPVConf>> {
        json::from_value(self.conf["spv_conf"].clone())
            .map_to_mm(|e| UtxoConfError::ErrorDeserializingSPVConf(e.to_string()))
    }
```
 any reason you changed this?

Reply:
probably a mistake while merging